docs/sample_log_config.yaml

```diff
@@ -18,13 +18,27 @@ filters:
 
 handlers:
     file:
-        class: logging.handlers.RotatingFileHandler
+        class: logging.handlers.TimedRotatingFileHandler
         formatter: precise
         filename: /var/log/matrix-synapse/homeserver.log
-        maxBytes: 104857600
-        backupCount: 10
-        filters: [context]
+        when: midnight
+        backupCount: 3  # Does not include the current log file.
         encoding: utf8
+
+    # Default to buffering writes to log file for efficiency. This means that
+    # will be a delay for INFO/DEBUG logs to get written, but WARNING/ERROR
+    # logs will still be flushed immediately.
+    buffer:
+        class: logging.handlers.MemoryHandler
+        filters: [context]
+        target: file
+        # The capacity is the number of log lines that are buffered before
+        # being written to disk. Increasing this will lead to better
+        # performance, at the expensive of it taking longer for log lines to
+        # be written to disk.
+        capacity: 10
+        flushLevel: 30  # Flush for WARNING logs as well
+
     console:
```

Comment:
if we're going to keep this, we document what it does and how to use it. I think it's by replacing `file` with `console` in both `buffer` and `twisted`.

Reply:
also, you might consider reordering, to put it next to `file` (since the two are alternatives)

Reply:
(or maybe we say it's an alternative to `buffer`? in which case you use it by replacing `buffer` and `file` below?)